pyccel/ast/core.py

```diff
@@ -3861,6 +3810,125 @@ def __str__(self):
                 args   = ', '.join(self.args),
                 result = result)
 
+class Interface(Basic):
+
+    """Represents an Interface.
+
+    Parameters
+    ----------
+    name : str
+        The name of the interface.
+
+    functions : iterable
+        The functions of the interface.
+
+    hide: bool
+        if True, the interface definition will not be generated.
+
+    is_argument: bool
+        True if the interface is used for a function argument.
+
+    Examples
+    --------
+    >>> from pyccel.ast.core import Interface, FunctionDef
+    >>> f = FunctionDef('F', [], [], [])
+    >>> Interface('I', [f])
+    """
+
+    def __new__( cls, *args, **kwargs ):
+        return Basic.__new__(cls)
+
+    def __init__(
+        self,
+        name,
+        functions,
+        hide=False,
+        is_argument = False,
+        ):
+
+        if not isinstance(name, str):
+            raise TypeError('Expecting an str')
+        if not isinstance(functions, list):
+            raise TypeError('Expecting a list')
+        self._name = name
+        self._functions = functions
+        self._hide = hide
+        self._is_argument = is_argument
+
+    @property
+    def name(self):
+        """Name of the interface."""
+        return self._name
+
+    @property
+    def functions(self):
+        """"Functions of the interface."""
+        return self._functions
+
+    @property
+    def hide(self):
+        """If True, the interface definition will not be generated."""
+        return self._functions[0].hide or self._hide
+
+    @property
+    def is_argument(self):
+        """True if the interface is used for a function argument."""
+        return self._is_argument
+
+    @property
+    def global_vars(self):
+        """Variables wich will not be passed to the interface."""
+        return self._functions[0].global_vars
+
+    @property
+    def cls_name(self):
+        """Class name if the interface is a method of cls_name."""
+        return self._functions[0].cls_name
+
+    @property
+    def kind(self):
+        """ 'function' or 'procedure' """
+        return self._functions[0].kind
```

Comment:
Yes I agree this shouldn't be `function or procedure`. this code was here before I didn't know if I should remove

Reply:
@EmilyBourne I don't know why this whole class is shown as being modified, I think git is just confused. @rmahjoubi made some changes to class `Interface`, but he did not rewrite it completely.

Reply:
Ah ok. I think he moved it in the file which is probably why it's all shown up as new. So I wasn't sure what was new and what wasn't or if it was just a rewrite. But I thought this might be a good opportunity to look at the contents of the class as we are unlikely to do it later

Reply:
I removed all unused methods in the class Interface